### PR TITLE
Fixes bluespace shelters destroying shuttles

### LINKED ISSUE #13842: Bluespace capsules break shuttles
Since they create a new area, the shuttle will have a big capsule house shaped hole in it when it moves

I guess you could argue FEATURE since you're making a fucking house in the shuttle but its kind of wonky and unbalanced.

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Fixes #13842

### CODE REVIEW

File: code/modules/mining/mine_items.dm [hunk 1]
```diff
@@ -243,7 +243,9 @@
 	threshhold.nitrogen = 82
 	threshhold.carbon_dioxide = 0
 	threshhold.toxins = 0
-	L.contents += threshhold
+	var/area/ZZ = get_area(threshhold)
+	if(!(istype(ZZ, /area/shuttle)))
```
Comment (tkdrg) [inline code/modules/mining/mine_items.dm]: Should probably make this an area blacklist instead of a snowflake istype()

Comment (KazeEspada) [inline code/modules/mining/mine_items.dm]: Blacklists are worse than snowflake.

Comment (duncathan) [inline code/modules/mining/mine_items.dm]: This is a blacklist, just with only one item
Anyway, a blacklist is fine for this. We don't typically add areas that would need to be blacklisted.

**Comment by KorPhaeron** [comment]
@tkdrg 

Updated

File: code/modules/mining/mine_items.dm [hunk 2]
```diff
@@ -243,7 +244,9 @@
 	threshhold.nitrogen = 82
 	threshhold.carbon_dioxide = 0
 	threshhold.toxins = 0
-	L.contents += threshhold
+	var/area/ZZ = get_area(threshhold)
+	if(!(ZZ.type in blacklist))
```
Comment (tkdrg) [inline code/modules/mining/mine_items.dm]: This won't work with subtypes of `/area/shuttle`. Use `if(!is_type_in_list(ZZ, blacklist))`

File: code/modules/mining/mine_items.dm [hunk 3]
```diff
@@ -261,4 +264,6 @@
 		A.air.temperature = 293.15
 		SSair.add_to_active(A)
 		A.overlays.Cut()
-		L.contents += A
+		var/area/Z = get_area(A)
+		if(!(Z.type in blacklist))
```
Comment (tkdrg) [inline code/modules/mining/mine_items.dm]: Same as above

**Comment by KorPhaeron** [comment]
@tkdrg 

Actually fixed

**Pull request merged**